src/main/java/org/folio/consortia/security/SecurityManagerService.java

```diff
@@ -0,0 +1,163 @@
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.folio.consortia.client.PermissionsClient;
+import org.folio.consortia.client.UsersClient;
+import org.folio.consortia.domain.dto.Permission;
+import org.folio.consortia.domain.dto.PermissionUser;
+import org.folio.consortia.domain.dto.Personal;
+import org.folio.consortia.domain.dto.SystemUserParameters;
+import org.folio.consortia.domain.dto.User;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.google.common.io.Resources;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Component
+@Log4j2
+@RequiredArgsConstructor
+public class SecurityManagerService {
+
+  private static final String PERMISSIONS_FILE_PATH = "permissions/system-user-permissions.csv";
+  private static final String USER_LAST_NAME = "SystemConsortia";
+
+  private final PermissionsClient permissionsClient;
+  private final UsersClient usersClient;
+  private final AuthService authService;
+
+  @Value("${folio.system.username}")
+  private String username;
+
+  @Value("${folio.system.password}")
+  private String password;
+
+  public void prepareSystemUser(String okapiUrl, String tenantId) {
+    Optional<User> userOptional = getUser(username);
+
+    User user;
+    if (userOptional.isPresent()) {
+      user = userOptional.get();
+      updateUser(user);
+    } else {
+      user = createUser(username);
+      authService.saveCredentials(SystemUserParameters.builder()
+        .id(UUID.randomUUID())
+        .username(username)
+        .password(password)
+        .okapiUrl(okapiUrl)
+        .tenant(tenantId)
+        .build());
+    }
+
+    Optional<PermissionUser> permissionUserOptional = permissionsClient.get("userId==" + user.getId())
+      .getPermissionUsers()
+      .stream()
+      .findFirst();
+    if (permissionUserOptional.isPresent()) {
+      addPermissions(permissionUserOptional.get());
+    } else {
+      createPermissionUser(user.getId());
+    }
+  }
+
+  private Optional<User> getUser(String username) {
+    return usersClient.getUsersByQuery("username==" + username)
+      .getUsers()
+      .stream()
+      .findFirst();
+  }
+
+  private User createUser(String username) {
+    var result = createUserObject(username);
+    log.info("Creating {}.", result);
+    usersClient.saveUser(result);
+    return result;
+  }
+
+  private void updateUser(User user) {
+    if (existingUserUpToDate(user)) {
+      log.info("{} is up to date.", user);
+    } else {
+      populateMissingUserProperties(user);
+      log.info("Updating {}.", user);
+      usersClient.updateUser(user.getId(), user);
+    }
+  }
+
+  private PermissionUser createPermissionUser(String userId) {
```

Comment:
This logic for permissions looks very similar to PermissionUserServiceImpl https://github.com/folio-org/mod-consortia/blob/ed1692a2d3ca5e938a0efc6b164676aeff7d5a91/src/main/java/org/folio/consortia/service/impl/PermissionUserServiceImpl.java#L22 , could you reuse it?

Reply:
resed PermissionService